src/persist-types/src/error.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Codec related errors.
+
+use std::{error, fmt};
+
+/// A Codec related error.
+#[derive(Debug, Clone)]
+pub enum CodecError {
```

Comment:
I went back and forth on whether this should be called `Error` and imported in as `CodecError` where needed